P2 2800 - Initial Setup of Employer Feedback Web

philhead1 opened the pull request.

philhead1 committed:
WIP

philhead1 committed:
Removed code to launch webpage

philhead1 committed:
Initial setup of Employer feedback web

ChrisJWoodcock commented on .gitignore:
```diff
@@ -348,3 +348,6 @@ MigrationBackup/
 .ionide/
+
+src/.idea/.idea.SFA.DAS.EmployerFeedback.Web/.idea/
+/src/SFA.DAS.EmployerFeedback.Web/Controllers/HomeController.cs
```
Is this ignoring the HomeController?

ChrisJWoodcock commented on README.md:
````diff
@@ -1,165 +1,57 @@
-    - Replayed - messages will be moved back onto the original processing queue so that they can be processed again
-    - Deleted - messages will be removed and will be no longer available for processing
-```
+This web solution is part of Request Apprentice Training (RAT) project. Here the employer users can search for apprentice training using the extended search facility and optionally create requests for training course.
````
It is not part of the RAT

ChrisJWoodcock commented on README.md:
```diff
@@ -1,165 +1,57 @@
-[![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=_projectId_&metric=alert_status)](https://sonarcloud.io/dashboard?id=_projectId_)
-[![Jira Project](https://img.shields.io/badge/Jira-Project-blue)](https://skillsfundingagency.atlassian.net/secure/RapidBoard.jspa?rapidView=564&projectKey=_projectKey_)
-[![Confluence Project](https://img.shields.io/badge/Confluence-Project-blue)](https://skillsfundingagency.atlassian.net/wiki/spaces/_pageurl_)
+[![Build Status](https://dev.azure.com/sfa-gov-uk/Digital%20Apprenticeship%20Service/_apis/build/status/das-employer-aan-web?branchName=main)](https://dev.azure.com/sfa-gov-uk/Digital%20Apprenticeship%20Service/_build/latest?definitionId=3244&branchName=main)
```
Not part of employer aan

ChrisJWoodcock commented on README.md:
```diff
@@ -1,165 +1,57 @@
-
-Data:
+### Config
+You can find the latest config file in [das-employer-config repository](https://github.com/SkillsFundingAgency/das-employer-config/blob/master/das-employer-request-apprentice-training/SFA.DAS.EmployerFeedback.Web.json)
```
Not in the rat folder

ChrisJWoodcock commented on pipeline-templates/job/deploy.yml:
```diff
@@ -30,18 +30,28 @@ jobs:
+            ApimResourceGroup: $(SharedApimResourceGroup)
+            ApimName: $(SharedApimName)
+            SubscriptionId: $(AppServiceName)
+            PipelineVariableName: EmployerRequestApprenticeTrainingApimSubscriptionKey
```
Wrong variable

ChrisJWoodcock commented on pipeline-templates/job/deploy.yml:
```diff
@@ -30,18 +30,28 @@ jobs:
-            ConfigurationSecrets: {} # remove this if there are no secrets that need to be mapped for config
+            ConfigurationSecrets: 
+              DefaultSessionRedisConnectionString: $(DefaultSessionRedisConnectionString)
+              EmployerRequestApprenticeTrainingApimSubscriptionKey: $(EmployerRequestApprenticeTrainingApimSubscriptionKey)
```
Wrong variable

ChrisJWoodcock commented on src/SFA.DAS.EmployerFeedback.Infrastructure/Api/Responses/Provider.cs:
```diff
@@ -0,0 +1,14 @@
+
+namespace SFA.DAS.EmployerFeedback.Infrastructure.Api.Responses
+{
+    public class Provider
```
Is this a placeholder or known to be correct - usually we use Ukprn to identity providers

philhead1 replied:
This is test code to test end to end retrieval of data from commitmentsV2

ChrisJWoodcock commented on src/SFA.DAS.EmployerFeedback.Infrastructure/Configuration/EmployerFeedbackWebConfiguration.cs:
```diff
@@ -0,0 +1,12 @@
+    [ExcludeFromCodeCoverage]
+    public class EmployerFeedbackWebConfiguration
+    {
+        public string FindApprenticeshipTrainingBaseUrl { get; set; }
```
It FAT used?

philhead1 replied:
Yes, fat is used to retrieve aggregated feedback stars and also aggregated feedback by UkPRN

ChrisJWoodcock commented on src/SFA.DAS.EmployerFeedback.Web/HealthChecks/ApiHealthCheck.cs:
```diff
@@ -0,0 +1,37 @@
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = new())
+        {
+            var description = "Ping of Employer Request Apprentice Training outer API";
```
Not RAT

ChrisJWoodcock commented on src/SFA.DAS.EmployerFeedback.Web/StartupExtensions/AuthorizationStartupExtensions.cs:
```diff
@@ -0,0 +1,37 @@
+        {
+            services.AddAuthorization(options =>
+            {
+                options.AddPolicy(PolicyNames.TransactorRole, policy =>
```
May need owner in here - but possible have to review stories

ChrisJWoodcock commented on src/SFA.DAS.EmployerFeedback.Web/Views/Home/Index.cshtml:
```diff
@@ -0,0 +1,24 @@
+﻿@using SFA.DAS.EmployerFeedback.Web.Controllers
+@{
+    ViewData["Title"] = "Employer Request Apprentice Training";
```
Update these not RAT

philhead1 committed:
Fixed issue with model not being displayed on test page

philhead1 committed:
Added back home controller as it was being ignored

philhead1 committed:
Addressing Review Comments

philhead1 committed:
Update .gitignore

Removed HomeController and .idea folder from gitignore

philhead1 committed:
Updates

Updated variable in deploy.yml and Also Added OwnerRole to AuthorizationStartupExtensions

philhead1 committed:
Updates for review

Added comment to remove Provider class as only used for testing connectivity

ChrisJWoodcock commented on src/SFA.DAS.EmployerFeedback.Web/Controllers/HomeController.cs:
```diff
@@ -0,0 +1,192 @@
+        {
+            _contextAccessor.HttpContext.Response.Cookies.Delete("SFA.DAS.EmployerFeedback.Web.Auth");
+        }
+#if DEBUG
```
Dashboard Stub and Overview Stub are for RAT I don't think there would be equivalents in Employer Feedback so you can delete them

philhead1 replied:
Removed 😄

philhead1 committed:
Update HomeController.cs

Removed DashboardStub and OverviewStub

debrr07 committed:
Removed stage specific variable groups from AT/TEST/TEST2/DEMO/PP

debrr07 committed:
Corrected Outer API product

debrr07 committed:
Commented out Generate Config step
